schema/rockcraft.json

```diff
@@ -619,7 +635,3251 @@
+              "build-environment": {
+                "title": "Build-Environment",
+                "default": [],
+                "type": "array",
+                "items": {
+                  "type": "object",
+                  "additionalProperties": {
+                    "type": "string"
+                  }
+                }
+              },
+              "build-attributes": {
+                "title": "Build-Attributes",
+                "default": [],
+                "type": "array",
+                "items": {
+                  "type": "string"
+                }
+              },
+              "organize": {
+                "title": "Organize",
+                "default": {},
+                "type": "object",
+                "additionalProperties": {
+                  "type": "string"
+                }
+              },
+              "overlay": {
+                "title": "Overlay",
+                "default": [
+                  "*"
+                ],
+                "type": "array",
+                "items": {
+                  "type": "string"
+                }
+              },
+              "stage": {
+                "title": "Stage",
+                "default": [
+                  "*"
+                ],
+                "type": "array",
+                "items": {
+                  "type": "string"
+                }
+              },
+              "prime": {
+                "title": "Prime",
+                "default": [
+                  "*"
+                ],
+                "type": "array",
+                "items": {
+                  "type": "string"
+                }
+              },
+              "override-pull": {
+                "title": "Override-Pull",
+                "type": "string"
+              },
+              "overlay-script": {
+                "title": "Overlay-Script",
+                "type": "string"
+              },
+              "override-build": {
+                "title": "Override-Build",
+                "type": "string"
+              },
+              "override-stage": {
+                "title": "Override-Stage",
+                "type": "string"
+              },
+              "override-prime": {
+                "title": "Override-Prime",
+                "type": "string"
+              },
+              "permissions": {
+                "title": "Permissions",
+                "default": [],
+                "type": "array",
+                "items": {
+                  "$ref": "#/definitions/Permissions"
+                }
+              }
+            },
+            "additionalProperties": false
+          }
+        },
+        {
+          "if": {
+            "properties": {
+              "plugin": {
+                "const": "make"
+              }
+            }
+          },
+          "then": {
+            "$comment": "common properties had to be repeated here or they would be considered invalid by the schema validator otherwise",
```

Comment:
what's the impact of the $comment field? This is more like an internal comment

Reply:
Yeah, it is an internal comment that has no impact on the schema. You asked me whether we can add comments to explain the repetitive part if anyone was to read the schema and wonder.

Reply:
Yeah I know, but I was referring to an actual `# comment`. But if you say this has no impact when using the schema, then it's alright

Reply:
Yeah, this is the official way for commenting in json schemas:
https://json-schema.org/understanding-json-schema/reference/comments